File: problems/1537/solution.cpp
class Solution {
public:
    int maxSum(vector<int>& a, vector<int>& b) {
        int i = 0, j = 0, m = a.size(), n = b.size();
        long long int sa = 0, sb = 0, mod = 1e9 + 7;
        
        while (i < m or j < n) {
            if (i < m and (j == n or a[i] < b[j])) {
                sa += a[i++];
            } else if (j < n and (i == m or a[i] > b[j])) {
                sb += b[j++];
            } else {
                sa = sb = max(sa, sb) + a[i];
                i++;
                j++;
            }
        }
        
        return max(sa, sb) % mod;
    }
};

File: problems/1537/solution.go
package main
const MOD = 1000000007

func maxSum(a []int, b []int) int {
    i, j, sa, sb, m, n := 0, 0, 0, 0, len(a), len(b)
    
    for i < m || j < n {
        if i < m && (j == n || a[i] < b[j]) {
            sa += a[i]
            i++
        } else if j < n && (i == m || a[i] > b[j]) {
            sb += b[j]
            j++
        } else {
            sa = max(sa, sb) + a[i]
            sb = sa
            i++
            j++
        }
    }
    
    return max(sa, sb) % MOD
}

func max(a, b int) int {
    if a > b {
        return a
    }
    
    return b
}